sdk/src/client_provider.rs

```diff
@@ -59,8 +61,15 @@ impl ClientProviderConfig {
                     client_address: args.quic_client_address,
                     server_address: args.quic_server_address,
                     server_name: args.quic_server_name,
-                    reconnection_retries: args.quic_reconnection_retries,
-                    reconnection_interval: args.quic_reconnection_interval,
+                    reconnection: TcpClientReconnectionConfig {
```

Comment:
move this to before line 58 `let reconnection = ...` and inside this struct just use reconnection